services/api-server/src/simcore_service_api_server/services/catalog.py

```diff
@@ -85,10 +90,10 @@ async def list_solvers(
         )
         response.raise_for_status()
 
+        services = parse_raw_as(list[TruncatedCatalogServiceOut], response.text)
```

Comment:
question here. this listing might be pretty big and parse_raw_as runs synchonously. are you sure this cannot be some kind of performance issue?

Reply:
Also I see you converted response.json() to response.text and then use parse_raw_as. Is this after measuring performance? is this an improvement?

Reply:
The only reason I did this is to try to make sure no unexpected exceptions can pop (so I am sure I handle all of them). It is not clear to me if the `.json()` method of the response can throw an exception. Hence, I parse it directly. I was guessing there would not be any performance issue since I guess the `.json` method also parses it under the hood. But TBH I have no clue. I definitely didn't profile this so I am happy revert this if we have a reason to suspect it introduces an overhead. Do you know @pcrespov?

Reply:
I can find [this](https://docs.pydantic.dev/dev/concepts/performance/#performance-tips) concerning pydantic's performance when parsing. There it suggests **not** first converting it to json as that is probably slower. Looking into the httpy source I can see that the `.json` method simply calls `json.loads`, so doubt there is a large performance overhead by letting pydantic do the parsing and probably there will be a performance improvement by doing it this way once we upgrade to pydantic 2. But ok that's based on what pydantic says. The proper way would be to profile it.

Reply:
yeah well my experience comes from the infamous catalog /v0/services endpoint that returns a _load_ of entries and parses them. And there I know that the parsing can be tough. Therefore if you expect a load of entries, I would definitely run this in an async executor ```await asyncio.get_event_loop.run_in_executor(None, validating_fct)```
but otherwise it's ok

Reply:
Just to be sure, are we talking about the difference between

```python
parse_obj_as(list[TruncatedCatalogServiceOut], response.json())
```
and
```python
parse_raw_as(list[TruncatedCatalogServiceOut], response.text)
```
?

If so, at first glance I would say that they are the same provided that pydantic uses the same [json_encoder](https://docs.pydantic.dev/1.10/usage/exporting_models/#json_encoders) (which i think it does)